src/Libraries/RevitNodes/Application/FamilyDocument.cs

```diff
@@ -0,0 +1,283 @@
+                    try
+                    {
+                        if (value.GetType() == typeof(int))
+                            doubleValue = Convert.ToDouble(value);
+                        else
+                            doubleValue = (double)value;
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOperationException(string.Format(Properties.Resources.WrongStorageType, familyParameter.StorageType));
+                    }
+                    
+                    var doubleValueToSet = doubleValue * UnitConverter.DynamoToHostFactor(familyParameter.Definition.UnitType);                                                           
+                    FamilyManager.Set(familyParameter, doubleValueToSet);
+                    break;
+
+                case Autodesk.Revit.DB.StorageType.String:
+                    string strValue;
+                    try
+                    {
+                        strValue = (string)value;
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOperationException(string.Format(Properties.Resources.WrongStorageType, familyParameter.StorageType));
+                    }
+                        
+                    FamilyManager.Set(familyParameter, strValue);
+                    break;
+
+                case Autodesk.Revit.DB.StorageType.ElementId:
+                    Autodesk.Revit.DB.ElementId id;
+                    try
+                    {
+                        id = ((Element)value).InternalElement.Id;
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOperationException(string.Format(Properties.Resources.WrongStorageType, typeof(Element)));
+                    }
+                    
+                    FamilyManager.Set(familyParameter, id);
+                    break;
+                default:
+                    return null;
+            }
+            TransactionManager.Instance.TransactionTaskDone();
+            return this;
+        }
+
+        /// <summary>
+        /// Add a new family parameter with a given name.
+        /// </summary>
+        /// <param name="parameterName">The name of the new family parameter.</param>
+        /// <param name="parameterGroup">The name of the group to which the family parameter belongs.</param>
+        /// <param name="parameterType">The name of the type of new family parameter.</param>
+        /// <param name="instance">Indicates if the new family parameter is instance or type.</param>
+        /// <returns>The new family parameter.</returns>
+        public FamilyParameter AddParameter(string parameterName, string parameterGroup, string parameterType, bool instance)
+        {
+            // parse parameter type
+            var type = Autodesk.Revit.DB.ParameterType.Text;
```

Comment:
do we need to declare this here ? could add the `Autodesk.Revit.DB.ParameterType` declaration right after `out` below (think `var` also works ;) )